Server/routing/index.js

```diff
@@ -27,39 +27,45 @@ router.get('/', async (ctx) => {
 
 router.get('/t/:miniURL', async (ctx) => {
     // find url in DB or error- does not exist
+    let id = parseInt(ctx.params.miniURL, 36);
+    console.log(id);
+    try {
+        var query = await database.query(
+            'SELECT * FROM public.urls WHERE id = ' + id
+        )
+
+        
+        if (query.rows[0] != null) {
+            await database.query(
```

Comment:
Let's add one more abstraction layer here. Instead of your handlers calling `database.query()` directly, add a "service" so the handler can just do `urlService.findOne(id)` and the service is responsible for calling `database.query()` and then returning the relevant record(s). You can leave the null checks and error handling in the handler if that's easiest.